### ISSUE TITLE
embedding in .net

### ISSUE THREAD

**Issue by user015** [pythonnet/pythonnet#644]
### Environment

-   Pythonnet version: 2.3.0
-   Python version: 3.6.3
-   Operating System: windows 7 x64

### Details
I am trying to embed python into my .NET application.  

These are steps I have done so far. 
a) downloaded and installed python ( 64bit)
b) pip pythonnet
c) included the Python.Runtime.dll in my solution. 
d) created pythonhome and pythonpath path variables.
e) downloaded the source code from github, got the zip file and unzipped it. trying to compile it from the command prompt, it fails with a error in the setup.py file about wheels, no module named wheel. 

The error is below

C:\Python.net\source\pythonnet-master>python setup.py bdist_wheel
Traceback (most recent call last):
  File "setup.py", line 18, in <module>
    from wheel import bdist_wheel
ModuleNotFoundError: No module named 'wheel'
                   (most recent call last):File "setup.py", line 18, in <module>from wheel import bdist_wheelM

We are looking at python.net due to IronPython's inability to work with numpy and scipy.  So can anyone can please guide me about what I am doing wrong ?

Thanks,

**Comment by denfromufa** [comment]
pip install wheel

**Comment by user015** [comment]
Thanks, that helped. But I still get errors during the compile process.  
This is the command I used 

python setup.py bdist_wheel

These are the errors I get 

CSC : error CS1617: Invalid option '6' for /langversion; must be ISO-1, ISO-2,
3, 4, 5 or Default [C:\Python.net\source\pythonnet-master\src\runtime\Python.Ru
ntime.csproj]
CSC : error CS1617: Invalid option '6' for /langversion; must be ISO-1, ISO-2,
3, 4, 5 or Default [C:\Python.net\source\pythonnet-master\src\clrmodule\clrmodu
le.csproj]

**Comment by denfromufa** [comment]
where Visual Studio versions do you have installed?

**Comment by user015** [comment]
visual studio professional 2012

**Comment by denfromufa** [comment]
you need vs 2015 or even better vs 2017

**Comment by denfromufa** [comment]
otherwise just grab the wheels from appveyor artifacts:

https://ci.appveyor.com/project/pythonnet/pythonnet/branch/master/job/4kafibai8l99cf22/artifacts

**Comment by user015** [comment]
Thanks a lot for your help. Will try these, the problem is our existing project has been using vs 2012, and It would be hard to move to 2017. I  will try out the artifacts.

**Comment by user015** [comment]
I downloaded the wheel and installed it using pip install <wheelname>
it went through fine and I was able to see it in the lib/sitepackages folder in python installation. 
I added a reference to this dll in my project and when I try to run my program, I get this error.

Unable to load DLL 'python36': The specified module could not be found. (Exception from HRESULT: 0x8007007E)

**Comment by denfromufa** [comment]
Please read the troubleshooting wiki

**Comment by yagweb** [comment]
@user015 Did you add the python path to the PATH variable? 

Please see this [code snippet](https://github.com/yagweb/pythonnetLab/blob/master/Test/Program.cs#L12-L41).

**Comment by user015** [comment]
Thanks @yagweb .  I have already done this, but has not worked.  
@denfromufa  I have also looked at the troubleshooting wiki. And still have the same error  

I am currently using a Form based .net project. would this be a issue ?

**Comment by user015** [comment]
C:\Python3.6\python.exe
Yes my project is set to 64 bit.  I am using the 64 bit wheel as well.  
At this point I have tried everything. nuget packages etc.

 All I can do now I would like to give this one final shot by starting off with a new project.

**Comment by denfromufa** [comment]
@user015 does this nPython.exe work on your machine? you can get it by compiling the pythonnet solution.

**Comment by user015** [comment]
Dont see any npython. 

 I removed python and starting fresh again.  Please let me know if any of these are not correct.

a) I installed latest python 3.6 (64 bit) from python.org. Will this version work or should I downgrade to a previous version ?

b) I am trying to install the 64 bit version of pythondotnet (pythonnet-2.3.0-cp36-cp36m-win_amd64.whl) using pip install command.
-or Should I install the 64 bit version ? I am on x64 not amd64  so which version should I install ?

**Comment by user015** [comment]
It works, deleting everything and reinstalling from Python worked.  The final steps seemed very simple,

a) installed latest python from python.org. 64 bit
b) pip install pythonnet
c) I referenced the python.runtime.dll from the pip installation folder.
d) It worked on .NET applications both Form based and console based
@denfromufa  Thanks a lot for your help.

I will explore the nuget way so that all of this can be easily packaged and not having to depend on user for installation. 

I got a unhandled Python.Runtime.PythonException at this location in the demo code, 
dynamic b = np.array(new List<float> { 6, 5, 4 }, dtype: np.int32)
This is the error message "TypeError : int() argument must be a string, a bytes-like object or a number, not '0, Culture=neutral, PublicKeyToken=b77a5c561934e089]]'"
